src/psd_channel.rs

```diff
@@ -149,14 +149,18 @@ pub trait IntoRgba {
 
         let mut idx = 0;
         let offset = channel_kind.rgba_offset().unwrap();
+        let len = cursor.get_ref().len() as u64;
 
-        while cursor.position() != cursor.get_ref().len() as u64 {
+        while cursor.position() < len {
             let header = cursor.read_i8() as i16;
 
             if header == -128 {
                 continue;
             } else if header >= 0 {
                 let bytes_to_read = 1 + header;
+                if cursor.position() + bytes_to_read as u64 > len {
+                    break;
+                }
```

Comment:
Tests are passing without this change.
What case are we solving for here?

Reply:
Yes, it is a real PSD file (and its owner has explicitly stated that it is confidential). It can be opened in ImageMagick and Krita, but FFmpeg also complains the file is malformed.
From its XMP metadata I can tell it had been edited by
Photoshop CC 2017 and then Photoshop 23.0 on Windows.

When the code hits this IF branch, `cursor.position()` and `len` are both 5964830, and `bytes_to_read` is 1. The 0 is at the end of the blue channel. I really have no idea why there is an extra byte.

Reply:
Got it, thanks for explaining.

I think we can unit test this then.

Can add something like this to the bottom of this file:

```rust
#[cfg(test)]
mod tests {
    use crate::PsdLayer;
    use super::*;

    /// Verify that when when inserting an RLE channel's bytes into an RGBA byte vec we do not
    /// attempt to read beyond the channel's length.
    #[test]
    fn does_not_read_beyond_rle_channels_bytes() {
        let layer = PsdLayer {
            channels: todo!("Set up some failing data"),
            layer_properties: todo!("Set up some failing data"),
        };
        
        let rgba = todo!("");
        let channel_bytes = todo!("");
        
        layer.insert_channel_bytes(&mut rgba, PsdChannelKind::Red, &channel_bytes);
    }
}
```

Then just craft a `PsdLayer` and `rgba` and `channel_bytes` that would fail without your fix.

---

Can also add a second test for the other fix below.

Reply:
Both tests have been added, please take another look.